dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml

```diff
@@ -4,109 +4,107 @@
-                        <th scope="col" class="govuk-table__header"></th>
-                    </tr>
-                </thead>
-                <tbody class="govuk-table__body">
-                    @foreach (var user in Model.Open!)
-                    {
-                        <tr class="govuk-table__row" data-testid="user-@user.UserId">
-                            <td class="govuk-table__cell"><a asp-page="User" asp-route-userId="@user.UserId">@user.Name</a></td>
-                            <td class="govuk-table__cell">@user.EmailAddress</td>
-                            <td class="govuk-table__cell"><a asp-page="AssignTrn/Trn" asp-route-userId="@user.UserId">Add a DQT record</a></td>
-                            <td class="govuk-table__cell"><a asp-page="AssignTrn/NoTrn" asp-route-userId="@user.UserId">Mark as non DQT</a></td>
-                        </tr>
-                    }
-                </tbody>
-            </table>
-        }
-    </govuk-tabs-item>
-    <govuk-tabs-item label="Closed" id="closed-tab" is-open="@true">
-        <p class="govuk-heading-s ">These records have been checked against the DQT.</p>
-
-        <p>Showing @Model.Closed!.Length of @Model.ClosedTotal total</p>
-
-        @if (Model.TotalPages > 1)
-        {
-            <govuk-pagination>
-                @if (Model.PreviousPage.HasValue)
+                @if (Model.TotalPages > 1)
                 {
-                    <govuk-pagination-previous asp-page="Users" asp-route-pageNumber="@Model.PreviousPage" asp-fragment="closed-tab" />
-                }
+                    <govuk-pagination>
+                        @if (Model.PreviousPage.HasValue)
+                        {
+                            <govuk-pagination-previous asp-page="Users" asp-all-route-data="@Model.GetPage((int)Model.PreviousPage!)" />
+                        }
 
-                @for (int i = 0; i < Model.PaginationPages!.Length; i++)
-                {
-                    var item = Model.PaginationPages[i];
+                        @for (int i = 0; i < Model.PaginationPages!.Length; i++)
+                        {
+                            var item = Model.PaginationPages[i];
 
-                    if (i > 0 && Model.PaginationPages[i - 1] != item - 1)
-                    {
-                        <govuk-pagination-ellipsis />
-                    }
+                            if (i > 0 && Model.PaginationPages[i - 1] != item - 1)
+                            {
+                                <govuk-pagination-ellipsis />
+                            }
 
-                    <govuk-pagination-item asp-page="Users" asp-route-pageNumber="@item" is-current="@(item == Model.PageNumber)" asp-fragment="closed-tab">@item</govuk-pagination-item>
-                }
+                            <govuk-pagination-item asp-page="Users" asp-all-route-data=@Model.GetPage(item) is-current="@(item == Model.PageNumber)">@item</govuk-pagination-item>
```

Comment:
Apparently you can't pass "page" in via asp-all-route-data. Throws a null pointer exception which had me confused for a while